src/Repository.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+namespace Friendica;
+
+use Friendica\Database\Database;
+use Friendica\Database\DBA;
+use Friendica\Network\HTTPException;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Repositories are Factories linked to one or more database tables.
+ *
+ * @see Model
+ * @see Collection
+ */
+abstract class Repository extends Factory
```

Comment:
The definitions aren't in this relation, but for easiness, it's fine :)

Reply:
What do you mean?

Reply:
A repository isn't a specialized factory, they are different patterns, but that falls under nudging ;)